frontend/apps/pablo/components/Organisms/staking/UnstakeForm/UnstakeModal.tsx

```diff
@@ -1,43 +1,46 @@
-import React from "react";
+import React, { useMemo } from "react";
 import { ModalProps, Modal } from "@/components/Molecules";
-import {
-  Box,
-  Typography,
-  useTheme,
-  Button,
-  Grid,
-} from "@mui/material";
+import { Box, Typography, useTheme, Button, Grid } from "@mui/material";
 
 import { useDispatch } from "react-redux";
 import { Label } from "@/components/Atoms";
 import { TokenValueItem } from "../TokenValueItem";
 import { TOKENS } from "@/defi/Tokens";
-import { setMessage } from "@/stores/ui/uiSlice";
-import { StakedFinancialNftPosition } from "@/defi/types";
+import { StakedFinancialNftPosition, StakingRewardPool } from "@/defi/types";
+import BigNumber from "bignumber.js";
 
 export type UnstakeModalProps = {
-  xPablo: StakedFinancialNftPosition,
+  stakingRewardPool?: StakingRewardPool;
+  xPablo: StakedFinancialNftPosition;
+  onDismiss: () => void,
+  onUnstake: () => void,
 } & ModalProps;
 
 export const UnstakeModal: React.FC<UnstakeModalProps> = ({
   xPablo,
-  onClose,
+  onDismiss,
+  onUnstake,
+  stakingRewardPool,
   ...modalProps
 }) => {
   const theme = useTheme();
   const dispatch = useDispatch();
 
-  const handleUnstake = () => {
-    onClose?.({}, "backdropClick");
-  };
+  const amount = useMemo(() => {
+    if (!stakingRewardPool) return new BigNumber(0);
+
+    return xPablo.isExpired
+      ? xPablo.lockedPrincipalAsset
+      : xPablo.lockedPrincipalAsset.minus(
+          xPablo.lockedPrincipalAsset.times(
+            stakingRewardPool.lock.unlockPenalty.div(100)
+          )
+        );
+  }, [xPablo, stakingRewardPool]);
 
   return (
-    <Modal
-      maxWidth="lg"
-      onClose={onClose}
-      {...modalProps}
-    >
-      <Box width={{md: 968}} margin="auto">
+    <Modal maxWidth="lg" onClose={onDismiss} {...modalProps}>
+      <Box width={{ md: 968 }} margin="auto">
```

Comment:
if this number is used elsewhere, it would be good to have it as a constant somewhere else.

Reply:
will create a ticket to update this later, makes more sense when updating to latest UI designs